internal/controllers/machinedeployment/machinedeployment_rollout_rollingupdate_test.go

```diff
@@ -596,6 +691,446 @@ func Test_reconcileOldMachineSetsRollingUpdate(t *testing.T) {
 	}
 }
 
+func TestReconcileInPlaceUpdateIntent(t *testing.T) {
+	utilfeature.SetFeatureGateDuringTest(t, feature.Gates, feature.InPlaceUpdates, true)
+
+	testCases := []struct {
+		name                   string
+		md                     *clusterv1.MachineDeployment
+		newMS                  *clusterv1.MachineSet
+		oldMS                  []*clusterv1.MachineSet
+		machines               []*clusterv1.Machine
+		scaleIntents           map[string]int32
+		upToDateResults        map[string]mdutil.UpToDateResult
+		canUpdateAnswer        map[string]bool
+		expectedCanUpdateCalls map[string]bool
+		expectMoveFromMS       []string
+		expectScaleIntents     map[string]int32
+	}{
+		// NewMS not scaling up
+
+		{
+			name:  "Move replicas from oldMS to newMS",
+			md:    createMD("v1", 2),
```

Comment:
> I think it doesn't matter, but maybe it's easy to fix, this MD should be v2, right?
(same/similar for other cases below)

I meant the version should be "v2" instead of "v1".

I see that you change the replica numbers. I'll review the replica changes